### DISCUSSION TITLE
Option to completely disable the new algorithmic feed (aka "Please don't turn GitHub into Facebook")

### DISCUSSION THREAD

**Discussion by mrjones2014** [community/community#13154] [Other Feature Feedback, Questions, & Ideas]
Please don't turn GitHub into Facebook.

Please give me an option to completely disable the algorithmic feed and only have the relevant, chronological feed, only from users I follow and repos I participate in.

**Comment by kLabz** [comment]
This. Now the information I am after is buried in a ton of "recommended for you" crap.

Edit: also, initial load is very slow, to the point I just go somewhere else before it even shows.

**Reply by mrjones2014** [reply]
Safari even did the thing where it said "This tab is using significant energy, closing it may improve performance" or whatever the wording is 😬 

Come on, GitHub. Big L.

**Reply by wangrongding** [reply]
# Github Old Feed  

💪 Using this script you can restore the old version of the feed.
https://github.com/wangrongding/github-old-feed  

<table>
  <tr><td width="50%" align="center"><b>Before (the shit💩 new feed)</b></td></tr>
  <tr>
     <td>
        <p align="center">
             <img width="800" align="center" src="https://assets.fedtop.com/picbed/1694066442147.jpg"/>
        </p>
     </td>
  </tr>
  <tr><td width="50%" align="center"><b>After use (the old one)</b></td></tr>
  <tr>
     <td>
        <p align="center">
             <img width="800" align="center" src="https://assets.fedtop.com/picbed/1694066437621.jpg"/>
        </p>
     </td>
  </tr>
</table>

**Reply by LunarWatcher** [reply]
Even when filtering out stuff, there's a whole bunch of useless crap included in the filters that I never asked for. Stars suddenly mean you follow releases to repos for example, and filtering that out filters out releases from watched repos where I explicitly wanted updates on releases. There's similar examples in all the other categories; tons of useless garbage in all the categories makes _all_ parts of the feed worthless and a pain to read through for the stuff I actually care about. 

See also https://github.com/orgs/community/discussions/66334

**Comment by bdresser** [comment]
Hey @mrjones2014, thanks for the feedback. 

>  only have the relevant, chronological feed, only from users I follow and repos I participate in.

The "following" tab only shows activity from users you follow and repositories you watch. This functionality is unchanged with the introduction of the new "For you" tab. Does this have what you're looking for, or am I misunderstanding your comment?

**Reply by mrjones2014** [reply]
Yeah, but when I go to the homepage, it defaults me to the algorithmic tab. So I want an option to completely disable it. It also causes significant performance issues.

**Reply by DianaOlympos** [reply]
yeah, let this be an opt-in option for people that wants it. Please. It is painful enough to deal with the tools we have when we are working on open source in our free time with limited resources. Slowing down that flow and making it one more click, and a click on a page that is slow af and a resource hog on top of that... wat

**Comment by slymax** [comment]


**Reply by mrjones2014** [reply]
Not really a solution, since I avoid Chrome/Chromium at all costs.

**Reply by suprstarrd** [reply]
@mrjones2014 A solution not being for you doesn't make it any less of a solution. I'm sure this can be ported to Firefox (or Safari or whatever browser you use).

**Reply by killshot13** [reply]
> Not really a solution, since I avoid Chrome/Chromium at all costs.

Even Brave Browser? 🤔

**Reply by Jawjomnissan** [reply]
https://github.com/community/community/discussions/13154#discussioncomment-2420337

**Comment by sntwandy** [comment]
I don't think it's a Facebook feed, for me it's very useful, it shows me interesting things about the repositories that I follow and about my colleagues in which they collaborate and others, as well as recommendations from other repositories.
Although he understands the fear behind the camera, at the moment I don't see that this is the focus, they are trying to make a feed more of global interaction than just in your circle.

**Comment by mrjones2014** [comment]
@bdresser any updates on this? I deleted my Facebook account years ago for a reason...

**Comment by mrjones2014** [comment]
Why does it default me to the algorithmic tab, even when I've manually gone back to the other tab? The algorithmic tab has horrible performance issues.

**Comment by chakravala** [comment]
Github will fade into irrelevance, just like twitter, a not serious place where big tech employees think they can manipulate what hits my eye balls for their own personal self-benefit

**Comment by lindhe** [comment]
I like the feature and think it's really useful, but I also think it is very important that it can be turned off. :)

**Reply by mrjones2014** [reply]
And also that I can opt out of having my data used for the algorithm.

**Reply by lindhe** [reply]
I fear GitHub won't the place for you then. :/ It's owned by Microsoft, we should expect too much in terms of privacy.

**Comment by alexitx** [comment]
Judging by my first impression (and expectations) I probably won't be touching such algorithm-driven feed ever. I mostly get recommended random users and organizations I have never interacted with and have nothing to do with my stars, forks and followed users. At least there could be a filter system to only show something like small and more obscure projects related to your starred repos and languages you work with.

**Comment by sirkon** [comment]
What is next? You will show "stories"?

**Reply by LinuxDragon57** [reply]
Typical Microsoft... buy a Technology. Promise not to ruin said technology. Proceed to ruin that technology anyway with useless garbage that their customers do not want: Skype, Halo (Cortana), Minecraft, Linkedin, and now Github. (There's probably more on that list). The scary part is that Microsoft owns the platform that hosts most of the world's Open-Source Software, and I think nothing speaks better to Microsoft's stance than its past: https://en.wikipedia.org/wiki/Criticism_of_Microsoft, specifically: https://en.wikipedia.org/wiki/Halloween_documents.

**Comment by LasseRafn** [comment]
I like the feature, BUT should be:
- [ ] Able to pick which one is default
- [ ] NOT loaded until I click into it; performance isn't great...

**Comment by LinuxDragon57** [comment]
I saw the stupid feed thing, and I was like wtf is this. Does Microsoft think Github is a social media platform?

**Comment by karolzlot** [comment]
Fortunately it's at least possible to use uBlock to remove feed completely.

Example rules:

```
! 2022-05-06 https://github.com
github.com###feed-next
github.com##.border-bottom.mt-3.col-lg-8.col-md-12
github.com##.border-bottom.hide-sm.hide-md.hide-lg.mt-5.col-lg-4.col-md-3.team-left-column
```


If you need to get some notifications it may make sense to create some program to get feed in RSS format and then e.g. notify when specific rules are met or just store this feed somewhere for later analysis.

**Reply by karolzlot** [reply]
@eladkarako Ok I removed them. Can you share what was broken by those rules?

**Reply by karolzlot** [reply]
Thank you, it is really helpful!

**Reply by kmk3** [reply]
Related thread:

* <https://github.com/orgs/community/discussions/53712#discussioncomment-5732926>

**Comment by alanxone** [comment]
**Stop turning Github into Facebook**, it's obviously a new product strategy after Microsoft acquired Github, but it's annoying.

GitHub SHOULD provide an option to disable such thing to give user the freedom of choice, and to let the organizations to decide if they want to enroll in this social feeds.

Rather than investing in these social algorithms, Github should consider improving team work experiences, especially for paying customers, such as bringing private team discussion into mobile app to make it useful, and enhancing the markdown wiki product to let the team publish changelog or knowledgebase in a more intuitive way.

**Comment by Ky6uk** [comment]
It might be good to have an option for "collaborators" in the feed filters. Sometime in the feed, I see a lot of unrelated crap with the reason You're seeing this because you collaborated with $userName.

I have worked with such users ages ago and my priorities/stack/etc. have been changed many times during this period of time. But for some reason, I'm just receiving noise about their activities.

**Comment by DianaOlympos** [comment]
I don't care about any of the information in there. I literally just unchecked all the boxes in the filters. I come to GitHub to work on a specific repository or follow up on my notifications. None of that is helped by this feed. It makes any interaction with Github when I come to work on it take at least one more click. Please let me get rid of it.

**Comment by zearp** [comment]
If you think Github will give you a choice or even remember the sorting setting in the session or as a user setting so you don't have to opt-out from this annoying sorting you're in for a surprise. Look at how the following simple request to control auto-updating of their desktop app has been lingering on for many years. A simple thing to do, wanted by many. Github just says its not on their roadmap to give us control and never bother with it again. They simply do not care. You want to just control how apps behave on your systems and networks? Haha, too bad! Github is not interested in what their customers and users want at all. They do not give a fsck.

https://github.com/desktop/desktop/issues/3410

**Reply by mrjones2014** [reply]
Pro tip, just don't use GitHub Desktop app, it's one of the worst git GUI clients out there anyway IMO. Or just do everything with git from command line, which imho is faster and easier anyway 😄

**Comment by infinitytec** [comment]
Everything relevant that I am interested in is in the Following feed. I don't need an algorithmic feed. 

I don't need to doomscroll on Github.

**Comment by aelgn** [comment]
Who are all these people? and what is all this noise?

**Comment by gpakosz** [comment]
I have no idea who these people are 🤷‍♂️

GitHub are you drunk?

<img width="428" alt="image" src="https://user-images.githubusercontent.com/553208/234913802-6e33e9d4-5ef0-42ac-9957-75eaca738e16.png">

**Comment by glinders** [comment]
I don't need the feed at all. If I want to know something I search for it.

How can I turn it off completely?

**Comment by mpaperno** [comment]
What happened... how did this become the ~~default~~ only view after hundreds of upvotes about how useless it is?

Let me just add in case upvoting isn't enough --- it's totally useless now!  I want my old feed back, please.

**Comment by PedroHLC** [comment]
If someone writes an userscript to make this new merged thing chronological and hide the stuff from people I don't follow, please let me know. I don't mind whether it has to fill my local-storage or do tons of requests in the background.

**Reply by fdwr** [reply]
And if anyone ever goes to the lengths of writing a script to improve user experience, that's strong evidence that the original design was flawed.

**Comment by vszakats** [comment]
Looks like the enlinkedinification of GitHub finally started after all.

**Comment by mryechkin** [comment]
I hate this new feed 🤮

**Reply by pschichtel** [reply]
"hate" doesn't even begin to describe what I think of it...

**Comment by cu** [comment]
This is what happens when Microsoft buys nice things.

**Reply by LinuxDragon57** [reply]
Absolutely what I've been thinking. Gosh I hate that company.

**Reply by LinuxDragon57** [reply]
Every time Microsoft buys something, they ruin it. It's a wonder how they still retain dominant market share in the desktop space given how shite their software is.

**Comment by wangrongding** [comment]
# Github Old Feed  

💪 Using this script you can restore the old version of the feed.
https://github.com/wangrongding/github-old-feed  

<table>
  <tr>
      <td width="50%" align="center"><b>Before</b></td>
      <td width="50%" align="center"><b>After use</b></td>
  </tr>
  <tr>
     <td><img src="https://assets.fedtop.com/picbed/1694066442147.jpg"/></td>
     <td><img src="https://assets.fedtop.com/picbed/1694066437621.jpg"/></td>
  </tr>
</table>

**Comment by mrjones2014** [comment]
They now [force For You onto everyone](https://github.com/orgs/community/discussions/66344)...

**Comment by mohkale** [comment]
For anyone now considering moving off GitHub I just found out about https://forgefed.org/ and https://forgejo.org/ which is a federated git instance. Make your own forge and avoid the centralization on GitHub which clearly no longer stands by what we originally hoped for.

**Comment by FMCalisto** [comment]
👋 Hello GitHub Community,

I've been actively using GitHub for my work in medical imaging and computer-assisted surgery. While I appreciate the platform's continuous efforts to enhance user experience, I share the concerns expressed by many here regarding the new algorithmic feed.

## 1. Feedback

1.1. **Relevance**: In my field, it's crucial to stay updated with the latest advancements and discussions. An algorithmic feed might not always capture the nuances of what's relevant to specialized domains like mine.

1.2. **Performance**: I've noticed a slowdown with the new feed, which can be a hindrance, especially when I'm trying to quickly navigate and catch up on updates.

1.3. **User Choice**: While innovations are always welcome, providing users the choice to opt-out or customize their feed would be beneficial. This ensures that the platform caters to both those who find the feature useful and those who prefer the traditional way.

## 2. Suggestions

2.1. **Customization**: Allow users to set preferences for their feed, choosing between algorithmic and chronological views.

2.2. **Performance Optimization**: Address the reported performance issues to ensure a smooth user experience.

2.3. **Feedback Loop**: Regularly gather feedback on new features to understand user preferences and make necessary adjustments.

Thank you for considering our feedback. GitHub has always been a cornerstone for developers, and I believe that with open communication and collaboration, we can continue to make it better for everyone.


Best regards,

@FMCalisto

**Comment by shailshouryya** [comment]
See https://github.com/orgs/community/discussions/66226 for a current (growing) list of related discussions

**Comment by mipmip** [comment]
I'm so sad. I hope a new GitHub will arise.

**Reply by mrjones2014** [reply]
GitLab or SourceHut
